ethereum/datastructures/src/main/java/tech/pegasys/artemis/datastructures/util/BeaconStateUtil.java

```diff
@@ -922,68 +547,24 @@ public static int bytes3ToInt(Bytes src, int pos) {
         | (src.get(pos + 2) & 0xFF);
   }
 
-  /**
-   * Splits ``values`` into ``split_count`` pieces.
-   *
-   * @param values The original list of validators.
-   * @param split_count The number of pieces to split the array into.
-   * @return The list of validators split into split_count pieces.
-   * @see <a
-   *     href="https://github.com/ethereum/eth2.0-specs/blob/v0.4.0/specs/core/0_beacon-chain.md#split">split
-   *     - Spec v0.4</a>
-   */
-  public static <T> List<List<T>> split(List<T> values, int split_count) {
-    checkArgument(split_count > 0, "Expected positive split_count but got %s", split_count);
-
-    int list_length = values.size();
-    List<List<T>> split_arr = new ArrayList<>(split_count);
-
-    for (int i = 0; i < split_count; i++) {
-      int startIndex = list_length * i / split_count;
-      int endIndex = list_length * (i + 1) / split_count;
-      List<T> new_split = new ArrayList<>();
-      for (int j = startIndex; j < endIndex; j++) {
-        new_split.add(values.get(j));
-      }
-      split_arr.add(new_split);
-    }
-    return split_arr;
-  }
-
-  /**
-   * Checks if the numerical value provided is a power of 2.
-   *
-   * @param value - The number under consideration.
-   * @return True if value is an exact power of 2, false otherwise.
-   * @see <a
-   *     href="https://github.com/ethereum/eth2.0-specs/blob/v0.4.0/specs/core/0_beacon-chain.md#is_power_of_two">is_power_of_two
-   *     - Spec v0.4</a>
-   */
-  public static boolean is_power_of_two(UnsignedLong value) {
-    long longValue = value.longValue();
-    return longValue > 0 && (longValue & (longValue - 1)) == 0;
-  }
-
   public static int get_beacon_proposer_index(
-      BeaconState state, UnsignedLong slot, boolean registry_change)
-      throws IllegalArgumentException {
-    UnsignedLong epoch = slot_to_epoch(slot);
-    UnsignedLong current_epoch = get_current_epoch(state);
-    UnsignedLong previous_epoch = get_previous_epoch(state);
-    UnsignedLong next_epoch = current_epoch.plus(UnsignedLong.ONE);
-
-    checkArgument(
-        previous_epoch.compareTo(epoch) <= 0 && epoch.compareTo(next_epoch) <= 0,
-        "get_beacon_proposer_index: slot not in range");
-
-    List<Integer> first_committee =
-        get_crosslink_committees_at_slot(state, slot, registry_change).get(0).getCommittee();
-    // TODO: replace slot.intValue() with an UnsignedLong value
-    return first_committee.get(epoch.intValue() % first_committee.size());
-  }
-
-  public static int get_beacon_proposer_index(BeaconState state, UnsignedLong slot) {
-    return get_beacon_proposer_index(state, slot, false);
+      BeaconState state) {
+    UnsignedLong epoch = get_current_epoch(state);
+    UnsignedLong committees_per_slot = UnsignedLong.valueOf(Math.floorDiv(get_epoch_committee_count(state, epoch).longValue(), (long)SLOTS_PER_EPOCH));
```

Comment:
You don't need to use Math.floorDiv when working with UnsignedLong.  The ONLY time floorDiv is different that integer division with truncation is when the integer is negative.  This can't happen with UnsignedLongs:
https://docs.oracle.com/javase/8/docs/api/java/lang/Math.html#floorDiv-long-long-

The code should be:
`UnsignedLong committees_per_slot = get_epoch_committee_count(state, epoch).dividedBy(UnsignedLong.valueOf(SLOTS_PER_EPOCH));`

Reply:
We do need to use floor other wise the rounding will happen at 0.5. Rounding need to be rounded down